lib/ui/components/large_button_widget.dart

```diff
@@ -0,0 +1,313 @@
+    required EnteColorScheme colorScheme,
+    required EnteColorScheme inverseColorScheme,
+  }) {
+    if (_isPrimary() || buttonType == ButtonType.critical) {
+      return strokeBaseDark;
+    }
+    if (_isSecondary()) {
+      return colorScheme.strokeBase;
+    }
+    if (buttonType == ButtonType.neutral ||
+        buttonType == ButtonType.trailingIcon) {
+      return inverseColorScheme.strokeBase;
+    }
+    if (buttonType == ButtonType.tertiaryCritical) {
+      return colorScheme.warning500;
+    }
+    //fallback
+    return colorScheme.strokeBase;
+  }
+
+  //Returning null to fallback to default color
+  Color? _pressedIconColor(EnteColorScheme colorScheme) {
+    if (buttonType == ButtonType.tertiaryCritical) {
+      return colorScheme.strokeBase;
+    }
+    return null;
+  }
+
+  //Returning null to fallback to default color
+  Color? _disabledIconColor(EnteColorScheme colorScheme) {
+    if (buttonType == ButtonType.primary ||
+        buttonType == ButtonType.secondary) {
+      return colorScheme.strokeMuted;
+    }
+    if (buttonType == ButtonType.critical ||
+        buttonType == ButtonType.tertiaryCritical) {
+      return colorScheme.strokeFaint;
+    }
+    return null;
+  }
+
+  TextStyle _defaultLabelStyle({
+    required EnteTextTheme textTheme,
+    required EnteTextTheme inverseTextTheme,
+  }) {
+    if (_isPrimary() || buttonType == ButtonType.critical) {
+      return textTheme.bodyBold.copyWith(color: textBaseDark);
+    }
+    if (_isSecondary()) {
+      return textTheme.bodyBold;
+    }
+    if (buttonType == ButtonType.neutral ||
+        buttonType == ButtonType.trailingIcon) {
+      return inverseTextTheme.bodyBold;
+    }
+    if (buttonType == ButtonType.tertiaryCritical) {
+      return textTheme.bodyBold.copyWith(color: warning500);
+    }
+    //fallback
+    return textTheme.bodyBold;
+  }
+
+  //Returning null to fallback to default color
+  TextStyle? _pressedLabelStyle(
+      EnteTextTheme textTheme, EnteColorScheme colorScheme) {
+    if (buttonType == ButtonType.tertiaryCritical) {
+      return textTheme.bodyBold.copyWith(color: colorScheme.strokeBase);
+    }
+    return null;
+  }
+
+  //Returning null to fallback to default color
+  TextStyle? _disabledLabelStyle(
+    EnteTextTheme textTheme,
+    EnteColorScheme colorScheme,
+  ) {
+    if (buttonType == ButtonType.primary ||
+        buttonType == ButtonType.secondary ||
+        buttonType == ButtonType.critical ||
+        buttonType == ButtonType.tertiaryCritical) {
+      return textTheme.bodyBold.copyWith(color: colorScheme.textFaint);
+    }
+    return null;
+  }
+
+  bool _hasTrailingIcon() {
+    return (buttonType == ButtonType.trailingIcon ||
+        buttonType == ButtonType.trailingIconPrimary ||
+        buttonType == ButtonType.trailingIconSecondary);
+  }
+
+  bool _isPrimary() {
+    return (buttonType == ButtonType.primary ||
+        buttonType == ButtonType.trailingIconPrimary);
+  }
+
+  bool _isSecondary() {
```

Comment:
When methods could be extension Methods on the enum..
Inside the button_type.dart enum file, can add

```lang=dart
extension ExtentendButtonType on ButtonType {
    bool isPrimary() {
      return this == ButtonType.primary || ButtonType.trailingIconPrimary
   }
}
```

Reply:
Will do this. In newer dart versions this can be done using enhanced enums. Can have a getter like `isPrimary` inside the enum and need not create an extension
method for it.